### PR TITLE
Fix missing permission check

### LINKED ISSUE #1422: 🪲 MT.1100 (Intune Diagnostic Settings) fails without showing Azure ARM permissions error
## Description

**MT.1100** (**Intune Diagnostic Settings should include Audit Logs**) needs a check in `public\maester\intune\Test-MtIntuneDiagnosticSettings.ps1` to see if the required Azure ARM permissions are present. If not, it should skip the test, passing the reason that it had insufficient permissions.

## Background

In my tenant, MT.1100 fails the test with the status, "**No Intune Diagnostic Settings found.**" This is odd, because I manually verified that I _do_ have Intune diagnostics all enabled and being shipped to Sentinel.

Maester's host output sometimes(?) shows _this_ error, which gave me a direction to start looking in:

```output
Select-Object: C:\Users\SamErde\Documents\PowerShell\Modules\Maester\2.0.3\public\maester\intune\Test-MtIntuneDiagnosticSettings.ps1:34
Line |
  34 |  … erty Content | ConvertFrom-Json | Select-Object -ExpandProperty value
     |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Property "value" cannot be found.
```

I found the issue when I ran the line manually in PowerShell:

`Invoke-AzRestMethod -Method GET -Path "/providers/microsoft.intune/diagnosticSettings?api-version=2017-04-01-preview"`

```output
StatusCode : 403
Content    : {
               "error": {
                 "code": "AuthorizationFailed",
                 "message": "The client 'Redacted UPN' with object id 'Redacted Object Id' does not have authorization to perform action 'microsoft.intune/diagnosticSettings/read' over scope '/providers/microsoft.intune' or the scope is invalid. If access was recently granted, please refresh your credentials."
               }
             }
Method     : GET
RequestUri : https://management.azure.com/providers/microsoft.intune/diagnosticSettings?api-version=2017-04-01-preview
Version    : 1.1
```

This resource does not rely on our Graph API scopes or delegated Intune permissions, but rather Azure Resource Manager role assignments that grant `microsoft.intune/diagnosticSettings/read` over scope `/providers/microsoft.intune`.

This can be resolved by granting the required assignment at the subscription level:

`New-AzRoleAssignment -ObjectId '{User, Security Principal, Managed Identity, or Group}' -RoleDefinitionName 'Reader' -Scope '/subscriptions/{SubscriptionId}'`

Or by creating a custom role and assigning it to the identity that performs the check. This is the better approach, as it follows the principle of least privilege.

```powershell
# Get the subscription ID and user ID from the current context. Change if necessary.
$SubscriptionId = "$((Get-AzContext).Subscription.Id)"
$UserId = (Get-AzADUser -UserPrincipalName (Get-AzContext).Account.Id).Id

$CustomRole = @{
    Name = 'Intune Diagnostic Settings Reader'
    Description = 'Can read Intune diagnostic settings only'
    Actions = @('microsoft.intune/diagnosticSettings/read')
    NotActions = @()
    AssignableScopes = @("/subscriptions/$SubscriptionId")
}

New-AzRoleDefinition -Role $CustomRole

# Assign the custom role at subscription level
New-AzRoleAssignment -ObjectId $UserId -RoleDefinitionName 'Intune Diagnostic Settings Reader' -Scope "/subscriptions/$SubscriptionId"
```

After this, the user may need to reconnect to Azure, and then the test will proceed successfully. Until then, a check should be added to the test and documentation updated to reflect this requirement.

### PR DESCRIPTION
**Pull request by nicolonsky**
# Description

Fixes #1422.

## Contribution Checklist

Before submitting this PR, please confirm you have completed the following:

- [x] 📖 Read the [guidelines for contributing](https://maester.dev/docs/contributing) to this repository.
- [x] 🧪 Ensure the build and unit tests pass by running `/powershell/tests/pester.ps1` on your local system.

### CODE REVIEW

File: powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1 [hunk 1]
```diff
@@ -31,6 +31,12 @@ function Test-MtIntuneDiagnosticSettings {
     try {
         Write-Verbose 'Retrieving Intune Diagnostic Settings status...'
         $diagnosticSettingsRequest = Invoke-AzRestMethod -Method GET -Path "/providers/microsoft.intune/diagnosticSettings?api-version=2017-04-01-preview"
+
+        # check whether the user has permissions to read diagnostic settings
+        if ($diagnosticSettingsRequest.StatusCode -eq 403) {
```
[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1]: @nicolonsky, this suggestion makes sense, but it also dilutes the specificity of your error handling. You might do `if '403' {report the permissions issue } else {report a general error}`.

File: powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.md [hunk 2]
```diff
@@ -1,6 +1,29 @@
 This test checks for the existence of Intune Diagnostic settings collecting Intune Audit Logs.
 
+#### Test Prerequisites
 
+For this test to run, the executing principal must have permissions to read Intune diagnostic settings in Azure (`microsoft.intune/diagnosticSettings/read` action). This typically requires at least the 'Monitoring Reader' or 'Reader' Azure role assigned at the subscription where the target Intune Log Analytics workspace or storage account resides.
```
[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.md]: Double check. Looks like a valid suggestion here as well.

File: powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1 [hunk 3]
```diff
@@ -62,6 +68,9 @@ function Test-MtIntuneDiagnosticSettings {
         }
         Add-MtTestResultDetail -Result $testResultMarkdown
         return [bool]($diagnosticSettings | Where-Object { $_.properties.logs | Where-Object { $_.category -eq 'AuditLogs' -and $_.enabled -eq $true } })
+    } catch [System.UnauthorizedAccessException] {
+        Add-MtTestResultDetail -SkippedBecause Custom -SkippedCustomReason 'Insufficient permissions to read Intune diagnostic settings in Azure.'
+        return $null
```
[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/maester365/maester/pull/1465#discussion_r2856600172)

File: powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1 [hunk 4]
```diff
@@ -62,6 +70,9 @@ function Test-MtIntuneDiagnosticSettings {
...
```
[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1:77]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/maester365/maester/pull/1465#discussion_r2924622998)

[the remark this answers is not in the document]
Comment (merill) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1:77]: Do we really need custom catch and write messages for each type? 

While it's helpful to distinguish, I feel there could be more types of exceptions and our tests will have more error handling code than the actual business logic.

I'm thinking we should just leave it with a generic catch all and the error message itself will have the details.

For this specific test, I think it's okay to leave it since it's already written.

[the remark this answers is not in the document]
Comment (SamErde) [reply powershell/public/maester/intune/Test-MtIntuneDiagnosticSettings.ps1:77]: Agreed. Copilot sometimes takes defensive code into edge case scenarios!
